src/kaleidoscope/hardware/ez/ErgoDox/ErgoDoxScannerDescription.h

```diff
@@ -0,0 +1,48 @@
+/* -*- mode: c++ -*-
+ * Kaleidoscope-Hardware-EZ-ErgoDox -- ErgoDox hardware support for Kaleidoscope
+ * Copyright (C) 2018-2019  Keyboard.io, Inc
+ *
+ * Based on QMK (commit 8cdb4a915)
+ *  (C) Jack Humbert, Erez Zukerman, Oleg Kostyuk
+ * Original QMK sources:
+ *  - keyboards/ergodox_ez/ergodox_ez.c
+ *  - keyboards/ergodox_ez/ergodox_ez.h
+ *  - keyboards/ergodox_ez/matrix.c
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#ifdef ARDUINO_AVR_ERGODOX
+
+#include <Arduino.h>
+
+#include "kaleidoscope/driver/ScannerDescription.h"
+
+namespace kaleidoscope {
+namespace hardware {
+namespace ez {
+
+struct ErgoDoxScannerDescription : public kaleidoscope::driver::ScannerDescription {
+  SCANNER_DESCRIPTION(14, 6);
+};
+
+}
+}
+}
+
+typedef kaleidoscope::hardware::ez::ErgoDoxScannerDescription::KeyAddr KeyAddr;
```

Comment:
I agree. This was placed here because `KeyAddr` was needed before the hardware header had a chance to define it. For example, the LED drivers need `KeyAddr`, and those are usually included before the hardware class is defined. Bit of a chicken-and-egg problem.

Mind you, this is one area I want to clean up and explore my options. For now, I went with the first thing that worked and produced code that isn't terribly ugly.

Reply:
Understandable. I stumbled upon similar chicken and egg problems in other places when I tried to restructure stuff for the virtual hardware. This would mean that a `KeyLEDs` controller that is part of the `LEDs` controller would need to gain knowledge from `KeyScannerDescription`, no?

Reply:
Pretty much, yes.